internal/cli/tiltfile_result.go

```diff
@@ -0,0 +1,104 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/windmilleng/tilt/pkg/logger"
+
+	"github.com/windmilleng/tilt/internal/analytics"
+	"github.com/windmilleng/tilt/internal/tiltfile"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+// arbitrary non-1 value chosen to allow callers to distinguish between
+// Tilt errors and Tiltfile errors
+const TiltfileErrExitCode = 5
+
+type tiltfileResultCmd struct {
+	fileName string
+}
+
+var _ tiltCmd = &tiltfileResultCmd{}
+
+type cmdTiltfileResultDeps struct {
+	tfl tiltfile.TiltfileLoader
+}
+
+func newTiltfileResultDeps(tfl tiltfile.TiltfileLoader) cmdTiltfileResultDeps {
+	return cmdTiltfileResultDeps{
+		tfl: tfl,
+	}
+}
+
+func newTiltfileResultCmd() *tiltfileResultCmd {
+	return &tiltfileResultCmd{}
+}
+
+func (c *tiltfileResultCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tiltfile-result",
+		Short: "Exec the Tiltfile and print results as JSON (note: the API is unstable and may change)",
+	}
+
+	cmd.Flags().StringVar(&c.fileName, "file", tiltfile.FileName, "Path to Tiltfile")
+
+	return cmd
+}
+
+func (c *tiltfileResultCmd) run(ctx context.Context, args []string) error {
+	// HACK(maia): we're overloading the -v|--verbose flags here, which isn't ideal,
```

Comment:
What's the expected behavior here? I'm having trouble inferring what you're trying to accomplish.

TBH I expected this to be as simple as logger.WithLogger(ctx, logger.NewLogger(logLvl, os.Stderr)); I couldn't figure out what the extra options are for.

Reply:
Oh, I'm just not printing Tiltfile logs except a. in verbose mode or b. on error. I don't feel strongly about this thi.

Reply:
let's print the logs to stderr all the time. If someone complains that it should be an option, we can add the option then. simplicity is good!

Reply:
ah whoops I missed this. how strongly do you feel about this? I think there's a strong case for "it's simpler for the user to have less output", and "only log stuff if requested or if something goes wrong" is a pretty common pattern, right?

mostly i'm not inclined to futz with this too much cuz it's alpha and i've already spent more time on it than i meant to, but 🤷‍♀